core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.spec.ts

```diff
@@ -325,6 +367,276 @@ describe('Contribution and review service', () => {
+          card_is_checkpoint: true
+        },
+        'End State': {
+          classifier_model_id: null,
+          content: {
+            content_id: 'content',
+            html: ''
+          },
+          recorded_voiceovers: {
+            voiceovers_mapping: {
+              content: {},
+              default_outcome: {}
+            }
+          },
+          interaction: {
+            answer_groups: [],
+            confirmed_unclassified_answers: [],
+            customization_args: {
+              recommendedExplorationIds: {
+                value: []
+              }
+            },
+            default_outcome: null,
+            hints: [],
+            solution: null,
+            id: 'EndExploration'
+          },
+          linked_skill_id: null,
+          param_changes: [],
+          solicit_answer_details: false,
+          card_is_checkpoint: false
+        }
+      };
+      const mockReadOnlyExplorationData = {
+        can_edit: true,
+        exploration: {
+          init_state_name: 'Introduction',
+          param_changes: [],
+          param_specs: {},
+          states: mockStates,
+          title: 'Dummy Title',
+          language_code: 'en',
+          objective: 'Dummy Objective',
+          next_content_id_index: '4',
+          correctness_feedback_enabled: true
+        },
+        exploration_metadata: {
+          title: 'Dummy Title',
+          category: 'Dummy Category',
+          objective: 'Dummy Objective',
+          language_code: 'en',
+          tags: [],
+          blurb: 'Dummy Blurb',
+          author_notes: 'Dummy Notes',
+          states_schema_version: 0,
+          init_state_name: 'Introduction',
+          param_specs: {},
+          param_changes: [],
+          auto_tts_enabled: true,
+          correctness_feedback_enabled: true,
+          edits_allowed: true,
+        },
+        exploration_id: '1',
+        is_logged_in: true,
+        session_id: '0',
+        version: 0,
+        preferred_audio_language_code: 'en',
+        preferred_language_codes: [],
+        auto_tts_enabled: true,
+        correctness_feedback_enabled: true,
+        record_playthrough_probability: 1,
+        draft_change_list_id: 1,
+        has_viewed_lesson_info_modal_once: false,
+        furthest_reached_checkpoint_exp_version: 0,
+        furthest_reached_checkpoint_state_name: '',
+        most_recently_reached_checkpoint_state_name: '',
+        most_recently_reached_checkpoint_exp_version: 1,
+        displayable_language_codes: ['en'],
+      };
+      const dummyExplorationBackendDict: ExplorationBackendDict = {
+        init_state_name: 'Introduction',
+        param_changes: [],
+        param_specs: {},
+        states: mockStates,
+        title: 'Dummy Title',
+        language_code: 'en',
+        objective: 'Dummy Objective',
+        correctness_feedback_enabled: true,
+        next_content_id_index: 4
+      } as unknown as ExplorationBackendDict;
```

Comment:
Removed

Reply:
@Lawful2002 Ah, actually we need this unknown to avoid assertion error, similar to what we have in https://github.com/oppia/oppia/blob/104e50f5a6467c14b4e2075e6c7bba9cdc2cf3c1/core/templates/pages/exploration-editor-page/services/exploration-data.service.spec.ts#L100